ibis/expr/window.py

```diff
@@ -10,6 +10,24 @@ def _sequence_to_tuple(x):
     return tuple(x) if util.is_iterable(x) else x
 
 
+def _determine_how(preceding):
+    if isinstance(preceding, tuple):
+        start, end = preceding
+        if start is None:
+            offset_type = type(end)
+        else:
+            offset_type = type(start)
+    else:
+        offset_type = type(preceding)
+
+    if isinstance(offset_type, type(int)):
```

Comment:
Since `int` is an instance of `type`, `type(int)` is equal to `type`. I think you mean just `int` here.

Reply:
This also needs to be `issubclass` since you're testing a `type` object (`offset_type`) as opposed to testing a value's type.

Reply:
I don't think there's a test that's causing this to fail. Can you try and add one?

Reply:
Added unit test and fixed isinstance/issubclass issue in next commit